packages/api/src/core/prepareParams.ts

```diff
@@ -336,7 +328,7 @@ export default async function prepareParams(operation: Operation, body?: unknown
           break;
         case 'header':
           params.header[paramName.toLowerCase()] = value;
-          delete metadata[paramName];
```

Comment:
We weren't properly cleaning out case-insensitive headers from here before, potentially resulting in some edge cases where `multipart/form-data` requests might not be handled properly.